### PR TITLE
drivers: kscan: Add STMPE811 I/O expander driver

### PR DESCRIPTION
**Pull request by papadkostas**
Add driver for ST STMPE811 I/O expander but only for touch operation and is based on ST BSP drivers.
Datasheet is available [here](https://media.digikey.com/pdf/Data%20Sheets/ST%20Microelectronics%20PDFS/STMPE811.pdf). The IC production is discontinued.
Added also support on the STM32F429I DISCO board and configuration for sample LVGL.
Driver tested succesfully on the board.

### CODE REVIEW

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: drivers/kscan/kscan_stmpe811.c
Comment (papadkostas) [reply drivers/kscan/kscan_stmpe811.c:168]: Well, numbers 240 and 320 are display size. How can I within this source file get the display size from DT?
The rest numbers are magic numbers nowhere explained, they are hard coded into BSP files from STMicro.

Comment (MaureenHelm) [reply drivers/kscan/kscan_stmpe811.c:168]: Have a look at the height and width properties defined in `dts/bindings/display/display-controller.yaml`, and their usage in `drivers/display/display_mcux_elcdif.c`

@erwango, can you help with the other magic numbers?

Comment (papadkostas) [reply drivers/kscan/kscan_stmpe811.c:168]: @MaureenHelm Thanks, will check it.

File: drivers/kscan/kscan_stmpe811.c
Comment (mbolivar-nordic) [inline drivers/kscan/kscan_stmpe811.c:2]: Missing >

File: drivers/kscan/kscan_stmpe811.c
Comment (mbolivar-nordic) [inline drivers/kscan/kscan_stmpe811.c:269]: What's going on here? Above, `cmd` is initialized to 0. Here, it's overwritten to this value, making the initialization a dead store. But the following line hard-codes STMPE811_REG_CHIP_ID again instead of reading `cmd`.

File: drivers/kscan/kscan_stmpe811.c
Comment (mbolivar-nordic) [inline drivers/kscan/kscan_stmpe811.c:327]: If this may vary by use case, this should come from devicetree instead.

File: drivers/kscan/kscan_stmpe811.c
Comment (mbolivar-nordic) [inline drivers/kscan/kscan_stmpe811.c:355]: A lot of this initialization seems like it might belong in DT, actually. I will leave it to the ST maintainers to comment further.

File: drivers/kscan/kscan_stmpe811.c
Comment (mbolivar-nordic) [inline drivers/kscan/kscan_stmpe811.c:432]: Even when CONFIG_KSCAN_STMPE811_INTERRUPT=y, the `int-gpios` property is still `required: false`. Therefore it might be missing from the devicetree. What is the right behavior in that case? What if there are 2 devices, one with an interrupt GPIO connected to the SoC, and one without? It seems like the driver may be able to handle this case without a compile error.

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]
Thanks for the patch.

This -1 is for:

- unclear division of work between driver and devicetree; this has already been commented on by @MaureenHelm but I see a couple of other places that are suspicious in that regard
- unclear handling of `int-gpios`, especially when there are multiple devices instances

Some other nitpicks also.

**Comment by papadkostas** [comment]
> Thanks for the patch.
> 
> This -1 is for:
> 
> * unclear division of work between driver and devicetree; this has already been commented on by @MaureenHelm but I see a couple of other places that are suspicious in that regard
> * unclear handling of `int-gpios`, especially when there are multiple devices instances
> 
> Some other nitpicks also.

Thanks for reviewing. I'll try to cover all the requirements mentioned from both.

**Pull request closed without merging**
